### PR TITLE
Shell Open With Default Program Master

### PR DESCRIPTION
**Pull request by fire**
4.0 edition of https://github.com/godotengine/godot/pull/39075

Will test for a few moments.

### CODE REVIEW

File: editor/filesystem_dock.cpp [hunk 1]
```diff
@@ -2749,6 +2763,17 @@ FileSystemDock::FileSystemDock(EditorNode *p_editor) {
 	file_list_display_mode = FILE_LIST_DISPLAY_THUMBNAILS;
 
 	always_show_folders = false;
+
+	shell_ignored_extensions.push_back("scn");
```
Comment (groud) [inline editor/filesystem_dock.cpp:2767]: Couldn't we compute that automatically? Or maybe get it from somewhere? 
I am pretty sure we will forget to maintain this list if we makes changes somewhere.

File: editor/filesystem_dock.cpp [hunk 2]
```diff
@@ -1575,6 +1575,17 @@ void FileSystemDock::_file_option(int p_option, const Vector<String> &p_selected
 			OS::get_singleton()->shell_open(String("file://") + dir);
 		} break;
 
+		case FILE_OPEN_IN_SHELL: {
```
Comment (groud) [inline editor/filesystem_dock.cpp:1578]: I find "OPEN_IN_SHELL" a little bit misleading, as it looks like it's going to open a terminal. OPEN_WITH_DEFAULT_PROGRAM would be better IMHO.

File: editor/filesystem_dock.cpp (hunk 1 above)
Comment (fire) [reply editor/filesystem_dock.cpp:2908]: Any suggestions how to do this automatically? Alternatively we can do it by whitelist.

**Comment by fire** [comment]
The last comment mentioned we should have a list of default extensions to block. This pr is salvageable.

**Pull request closed without merging**